src/backend/opencl/convolve.cpp

```diff
@@ -103,5 +112,184 @@ INSTANTIATE(ushort, float)
 INSTANTIATE(short, float)
 INSTANTIATE(uintl, float)
 INSTANTIATE(intl, float)
+#undef INSTANTIATE
+
+template <typename T, typename accT>
+Array<T> convolve2_unwrap(const Array<T> &signal, const Array<accT> &filter,
+                          const dim4 stride, const dim4 padding,
+                          const dim4 dilation) {
+    dim4 sDims = signal.dims();
+    dim4 fDims = filter.dims();
+
+    dim_t outputWidth =
+        1 +
+        (sDims[0] + 2 * padding[0] - (((fDims[0] - 1) * dilation[0]) + 1)) /
+            stride[0];
```

Comment:
This seems like an odd kind of spacing/indentation. I suppose these were automatically done by clang-format?

Reply:
yup